doc/Data Plane Code Generation/DeveloperDrivenEvolution.md

```diff
@@ -0,0 +1,195 @@
+```
+### Replace output value
+- Replace the return value `Response` with `Response<T>` (`T` is your wanted output type, e.g., `MetricFeedback`).
+
+After these steps, your convenience method signature will be like:
+```C#
+public virtual async Task<Response<MetricFeedback>> CreateMetricFeedbackAsync(MetricFeedback feedback, CancellationToken cancellationToken = default);
+```
+### Rename method if needed
+Renaming method should follow the rules:
+
+**It is needed if ambiguous call exists**
+
+This situation happens when your protocol method and convenience method have the same required paramater list. E.g.,
+```C#
+// protocol method
+public virtual async Task<Response> GetMetricFeedbackAsync(string feedId, RequestContext context = null);
+// Convenience method
+public virtual async Task<Response<MetricFeedback>> GetMetricFeedbackAsync(string feedId, CancellationToken cancellationToken = default);
+```
+Ambiguous call exists when you call
+```C#
+GetMetricFeedbackAsync(feedId);
+```
+In this situation, you have to change the convenience name.
+
+**Suggested rename for most cases**
+
+If your convenience method has the same parameter list as that of the protocol method, we suggest 
+- Adding a suffix `Value` to the initial method name as the convenience method name, if the initial method is singular (e.g., change `GetMetricFeedbackAsync` to `GetMetricFeedbackValueAsync`).
+```C#
+public virtual async Task<Response<MetricFeedback>> GetMetricFeedbackValueAsync(string feedId, CancellationToken cancellationToken = default);
+```
+- Adding a suffix `Values` to the initial method name as the convenience method name, if the initial method is plural (e.g., change `GetMetricFeedbacksAsync` to `GetMetricFeedbackValuesAsync`).
+```C#
+public virtual async Task<Response<MetricFeedback>> GetMetricFeedbackValuesAsync(string feedId, CancellationToken cancellationToken = default);
+```
+**If this name doesn't make sense, pick the best name**
+
+If the suggested rename does not apply to your scenario, you could pick the best name fitting your API.
+## Create the models
+Currently the models should be created manually. We will support generate it automatically.
+
+After you create the models, you also need to add the helper methods mapping between raw response and model. Examples for model `MetricFeedback` (Models/MetricFeedback/MetricFeedback.cs) are:
+```C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricFeedback
+    {
+        // Mapping raw response to model
+        internal static MetricFeedback FromResponse(Response response)
+        {
+            using var document = JsonDocument.Parse(response.Content);
+            // Add your deserialization logic in DeserializeMetricFeedback
+            return DeserializeMetricFeedback(document.RootElement);
+        }
+
+        // Mapping model to raw request
+        internal static RequestContent ToRequestContent(MetricFeedback metricFeedback)
```

Comment:
Do we need to write 2 conversion methods for each model?   Can we leverage template to reduce the effort?

Reply:
We do need one for serialization and one for deserialization.